virttest/utils_net.py

```diff
@@ -1697,20 +1697,36 @@ def restart_guest_network(
     :param timeout: timeout value for command.
     """
     if os_type == "linux":
+        dhcp_clients = [("dhclient", "-r"), ("dhcpcd", "-k")]
+        dhcp_cmd = None
+        release_cmd = None
+
+        for client, release_arg in dhcp_clients:
+            try:
+                session.cmd(("which %s" % client))
+            except aexpect.ShellCmdError:
+                continue
+            dhcp_cmd = client
+            release_cmd = "%s %s" % (client, release_arg)
+            break
+
+        if dhcp_cmd is None:
+            raise Exception("No DHCP client found")
```

Comment:
```suggestion
            raise LookupError("No DHCP client found")
```
`LookupError` can be used for more generic lookup failure handling.